src/d3a/models/area/event_dispatcher.py

```diff
@@ -73,34 +82,33 @@ def broadcast_balancing_market_cycle(self, **kwargs):
     def broadcast_callback(self):
         return self._broadcast_notification
 
-    def _broadcast_notification(self, event_type: Union[MarketEvent, AreaEvent], **kwargs):
-        if not self.area.events.is_enabled and \
-           event_type not in [AreaEvent.ACTIVATE, AreaEvent.MARKET_CYCLE]:
-            return
-        # Broadcast to children in random order to ensure fairness
-        for child in sorted(self.area.children, key=lambda _: random()):
-            child.dispatcher.event_listener(event_type, **kwargs)
-        # Also broadcast to IAAs. Again in random order
-        for time_slot, agents in self._inter_area_agents.items():
-            if time_slot not in self.area._markets.markets:
+    def _broadcast_notification_to_agents_of_market_type(
+            self, market_type: AvailableMarketTypes, event_type: AreaEvent, **kwargs):
+        for time_slot, agents in self._get_agents_for_market_type(self, market_type).items():
+            if time_slot not in self.area._markets.get_market_instances_from_class_type(
```

Comment:
Here we're accessing a private property of a different class (`area._markets`). We do the same in several other modules, so I think that `area._markets` wants to be made public.

Reply:
I would rather do this in isolation after this PR is merged, because this touches many parts of the code and is not related with the PR itself